SAFIR updates

SomayehSaghamanesh opened the pull request:
## Changes in this pull request
Added two models of SAFIR scanners (generic geometry) to stir's scanners list.
Added two utilities to convert and trim projections.

## Testing performed
STIR-6 build tests were performed successfully without any error. Tests for SAFIR I/II data conversion and image reconstructions were performed successfully.

## Related issues
issue #1501

## Checklist before requesting a review
  - [x] I have performed a self-review of my code
  - [] I have added docstrings/doxygen in line with the guidance in the developer guide
  - [x] I have implemented unit tests that cover any new or modified functionality (if applicable)
  - [x] The code builds and runs on my machine
  - [] `documentation/release_XXX.md` has been updated with any functionality change (if applicable)

KrisThielemans commented:
Not sure why the `diff` is so large after the merge. @robbietuk saw something similar. @casperdcl is there anything going wrong with GitHub ATM?

It would be best to locally rebase your branch on `origin/master`, and force-push to here, or even create a new PR. sorry

Truncating sinograms can already be done via `SSRB`. Do we really need another utility?

SomayehSaghamanesh commented:
I haven't tried SSRB for our purposes. Does it work with generic geometries as well? 
Can I rebase and then continue in the same PR?

KrisThielemans commented:
> I haven't tried SSRB for our purposes. Does it work with generic geometries as well?

it won't work for span, but (could be made to) work for trimming and possibly view-mashing

> Can I rebase and then continue in the same PR?

Normally you can, but you will have to force push. Who knows what GitHub will do for the diff now, but worth a try.

casperdcl commented:
the first commit in this PR (59b812d) added `build_safir_updates` which looks like a build output directory to me...

KrisThielemans commented:
ok. didn't even see that!

@SomayehSaghamanesh , please remove your build (don't put it in the source dir, or add it to your local `.gitignore`), rebase, and force-push.

SomayehSaghamanesh commented:
But I thought I had rebased and removed that commit. Are you sure it
doesn't belong to another branch?

On Wed, Sep 4, 2024, 17:53 Kris Thielemans ***@***.***> wrote:

> ok. didn't even see that!
>
> @SomayehSaghamanesh <https://github.com/SomayehSaghamanesh> , please
> remove your build (don't put it in the source dir, or add it to your local
> .gitignore), rebase, and force-push.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/UCL/STIR/pull/1502#issuecomment-2329437963>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXA62EOMJF3B5HODTDPRHBTZU4UJNAVCNFSM6AAAAABNULAYIGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMRZGQZTOOJWGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

SomayehSaghamanesh commented:
I did several rebase and looks something went wrong in between. I'll take
care of this.

On Wed, Sep 4, 2024, 18:13 Kris Thielemans ***@***.***> wrote:

> clearly not. 5000+ files difference...
>
> if you want to drop a commit, easiest is to use interactive rebase I'm
> afraid.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/UCL/STIR/pull/1502#issuecomment-2329479071>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXA62EIDTSEXRWD3OPIWFZLZU4WTNAVCNFSM6AAAAABNULAYIGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMRZGQ3TSMBXGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

SomayehSaghamanesh commented:
I think the git problem is fixed now.

robbietuk commented on src/utilities/trim_projdata.cxx:
```diff
@@ -0,0 +1,246 @@
+   trim_projdata [-t num_tang_poss_to_trim] \
+        output_filename input_projdata_name 
+   \endcode
+   \param num_tang_poss_to_trim has to be smaller than the available number
+      of tangential positions.
```
It's not clear from this documentation which tangential positions are trimmed, min or max tangential positions?

SomayehSaghamanesh replied:
The trim should be symmetric. I can include in the documentation. 
Yes, two utilities should be included in the CMakeLists indeed. just ignored to be tracked locally. I'll do it.

robbietuk reviewed (changes requested):
Just a couple of comments. Trimming projdata conversation turned up in https://github.com/UCL/STIR/pull/1499#issuecomment-2330811164.

I think the two utilities need to be added to https://github.com/UCL/STIR/blob/master/src/utilities/CMakeLists.txt
